### PR TITLE
add `empty` attribute, better error for `timeseries()` on empty dataframe

### LINKED ISSUE #291: `timeseries()` on empty `IamDataFrame` fails very loudly
# Description of the issue

When calling `timeseries()` on an empty `IamDataFrame`, the internally used `pandas.DataFrame.pivot_table()` fails with a long, non-intuitive error message (see below). This might be quite intimidating to non-expert users.

## Improved behaviour

Checking whether the `data` table is empty before doing the pivot operation is straightforward. 

Two questions on implementation and expected behaviour, though:

1) Should `pyam` only write a message to the log file (as is done for `filter()`, see [here](https://github.com/IAMconsortium/pyam/blob/103b60f8f2f46fddae4d0325cbc2651cb098030a/pyam/core.py#L1001), or should a (nicer) error be raised?

2) If only a warning is written to the log, should the returned object be `None` or an empty `pandas.DataFrame` (but with the correct index dimensions).

# Error message

```
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
<ipython-input-48-36a29f3c3d4a> in <module>
----> 1 df.filter(variable='foo').timeseries()

~/Github/pyam/pyam/core.py in timeseries(self, iamc_index)
    382         df = (
    383             self.data
--> 384             .pivot_table(index=index, columns=self.time_col)
    385             .value  # column name
    386             .rename_axis(None, axis=1)

~/anaconda3/lib/python3.7/site-packages/pandas/core/generic.py in __getattr__(self, name)
   5177             if self._info_axis._can_hold_identifiers_and_holds_name(name):
   5178                 return self[name]
-> 5179             return object.__getattribute__(self, name)
   5180 
   5181     def __setattr__(self, name, value):

AttributeError: 'DataFrame' object has no attribute 'value'
```

### PR DESCRIPTION
**Pull request by danielhuppmann**
# Please confirm that this PR has done the following:

- [x] Tests Added
- [x] Documentation Added
- [ ] Description in RELEASE_NOTES.md Added

## Adding to RELEASE_NOTES.md (remove section after adding to RELEASE_NOTES.md)

Please add a single line in the release notes similar to the following:

```
- (#XX)[http://link-to-pr.com] Added feature which does something
```

# Description of PR

This PR adds an `empty` attribute and adds a better error message when doing `timeseries()` on an empty `IamDataFrame`.

Closes #291

### CODE REVIEW

File: pyam/core.py [hunk 1]
```diff
@@ -148,6 +148,11 @@ def tail(self, *args, **kwargs):
         """Identical to pd.DataFrame.tail() operating on data"""
         return self.data.tail(*args, **kwargs)
 
+    @property
+    def empty(self):
+        """Indicator whether ``IamDataFrame`` is empty"""
```
Comment (znicholls) [inline pyam/core.py]:
```suggestion
        """Is the ``IamDataFrame`` empty?"""
```

File: pyam/core.py [hunk 2]
```diff
@@ -380,7 +385,17 @@ def timeseries(self, iamc_index=False):
...
+
+        Raises
+        ------
+        AttributeError
+            ``IamDataFrame`` is empty
+        ValueError
+            reducing to IAMC-index yields an index with duplicates
         """
+        if self.empty:
+            raise AttributeError('this `IamDataFrame` is empty')
```
Comment (znicholls) [inline pyam/core.py]: Should this be an attribute error or perhaps a value error instead?

File: pyam/core.py (hunk 1 above)
Comment (danielhuppmann) [reply pyam/core.py]: I was using the pandas docstring as a template, see https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.DataFrame.empty.html

File: pyam/core.py (hunk 2 above)
Comment (danielhuppmann) [reply pyam/core.py]: done

**Comment by znicholls** [comment]
Oh nice then keep it as is, I was just doing what seemed to  read more
easily to me but consistency with the wider community is a better choice

**Comment by danielhuppmann** [comment]
thanks @znicholls for the review!

**Comment by danielhuppmann** [comment]
Merging as approved by @znicholls

**Pull request merged**